test/test.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+require('should');
+
+var async = require('async');
+var mongoose = require('mongoose');
+var shortIdModule = require('shortid');
+
+var ShortId = require('../index');
+
+
+mongoose.connect('mongodb://localhost/test');
+
+var Schema = mongoose.Schema;
+
+var defaultSchema = new Schema({
+    _id: ShortId,
+    num: Number
+});
+
+var alphaSchema = new Schema({
+    _id: {
+        type: ShortId,
+        len: 2,
+        alphabet: 'abc',
+        retries: 10
+    },
+    num: Number
+});
+
+var generatorSchema = new Schema({
+    _id: {
+        type: ShortId,
+        generator: function(options, callback) {
+            callback(null, ''+Date.now());
+        }
+    },
+    num: Number
+});
+
+var shortIdModuleSchema = new Schema({
+    _id: {
+        type: ShortId,
+        generator: function(options, callback) {
+            callback(null, shortIdModule.generate());
+        }
+    },
+    num: Number
+});
+
+var DefaultDoc = mongoose.model('defaultdoc', defaultSchema);
+var AlphaDoc = mongoose.model('alphadoc', alphaSchema);
+var GeneratorDoc = mongoose.model('generatordoc', generatorSchema);
+var ShortIdModuleDoc = mongoose.model('shortidmoduledoc', shortIdModuleSchema);
+
+describe('shortid', function() {
+
+    this.timeout(10000);
+
+    before(function(done) {
+
+        async.each([DefaultDoc, AlphaDoc, GeneratorDoc, ShortIdModuleDoc], function(model, eachNext) {
+            model.remove(eachNext);
```

Comment:
Removes all items in the collection; collection if it exists will still be left (with any defined indexes, etc)